### ISSUE TITLE
sys-libs/readline, app-shells/bash LTO patch no longer applies

### ISSUE THREAD

**Issue by monoidic** [InBetweenNames/gentooLTO#226]
As of sys-libs/readline-8.0 and app-shells/bash-5.0, the LTO patches doesn't apply, as the volatile keyword has been added to "int nsigint" in aclocal.m4 and configure.
Is volatile int good enough or is volatile sig_atomic_t needed?

**Comment by ThGravo** [comment]
I tried simply removing the patch and it compiles...

**Comment by ThGravo** [comment]
Also `/var/lib/layman/lto-overlay/sys-config/ltoize/files/patches/app-shells/bash/lto.patch` doesn't apply anymore.

**Comment by monoidic** [comment]
Ah, you're right. The same exact change was made in bash as well.

**Comment by InBetweenNames** [comment]
This is excellent news.  I'll remove the patches as soon as I'm able.

Regarding your question about `sig_atomic_t`, according to the C standard, the only correct way is using `volatile sig_atomic_t`.  However, that test can't use `sig_atomic_t`, because the test preceding it is the one that tests if your system even has `sig_atomic_t`.  So, they compromised and chose `volatile int` instead, which is enough to at least prevent the compiler from optimizing the write out.  However, the code still invokes undefined behaviour as per the C standard.  The good news is that on all modern systems, `sig_atomic_t` is just typedef'd to `int` anyway, so it makes literally no difference for our use cases.

TL;DR: `volatile int` is good enough.

**Comment by bemeurer** [comment]
@InBetweenNames Can a fix for this please be merged? It kind of breaks world updates

**Comment by InBetweenNames** [comment]
Should be fixed.  Thanks!
